src/main/java/org/jitsi/videobridge/eventadmin/callstats/ConferencePeriodicRunnable.java

```diff
@@ -0,0 +1,153 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jitsi.videobridge.eventadmin.callstats;
+
+import org.jitsi.service.neomedia.*;
+import org.jitsi.service.neomedia.stats.*;
+import org.jitsi.stats.media.*;
+import org.jitsi.util.*;
+import org.jitsi.videobridge.*;
+
+import java.util.*;
+
+/**
+ * Extends a {@link AbstractStatsPeriodicRunnable} which periodically generates
+ * a statistics for the conference channels.
+ *
+ * @author Damian Minkov
+ */
+public class ConferencePeriodicRunnable
+    extends AbstractStatsPeriodicRunnable<Conference>
+{
+    /**
+     * The <tt>Logger</tt> used by the <tt>ConferencePeriodicRunnable</tt>
+     * class and its instances to print debug information.
+     */
+    private static final Logger logger
+        = Logger.getLogger(ConferencePeriodicRunnable.class);
+
+    /**
+     * The {@link MediaType}s for which we will report to callstats.
+     */
+    private static final MediaType[] MEDIA_TYPES
+        = { MediaType.AUDIO, MediaType.VIDEO };
+
+    /**
+     * Constructs <tt>ConferencePeriodicRunnable</tt>.
+     * @param conference the conference.
+     * @param period the reporting interval.
+     * @param statsService the StatsService to use for reporting.
+     * @param conferenceIDPrefix prefix to use when creating conference IDs.
+     * @param initiatorID the id which identifies the current bridge.
+     */
+    ConferencePeriodicRunnable(
+        Conference conference,
+        long period,
+        StatsService statsService,
+        String conferenceIDPrefix,
+        String initiatorID)
+    {
+        super(conference,
+            period,
+            statsService,
+            conference.getName(),
+            conferenceIDPrefix,
+            initiatorID);
+    }
+
+    @Override
+    protected Map<String, Collection<? extends ReceiveTrackStats>>
+        getReceiveTrackStats()
+    {
+        return getTrackStats(true);
+    }
+
+    @Override
+    protected Map<String, Collection<? extends SendTrackStats>>
+        getSendTrackStats()
+    {
+        return getTrackStats(false);
+    }
+
+    /**
+     * Get stats receive or send from current conference.
+     *
+     * @param receive whether to get receive if <tt>true</tt> or
+     * send statistics otherwise.
+     * @param <T> the type of result stats Collection, ReceiveTrackStats
+     * or SendTrackStats.
+     * @return the result collection of stats grouped by endpointID.
+     */
+    private <T extends Collection> Map<String, T> getTrackStats(boolean receive)
```

Comment:
Why do we need "T" here? Can't we just return a "Map<String, Collection>"?

Reply:
Nevermind, it seems that it is needed.

Reply:
This is how the abstract methods of AbstractStatsPeriodicRunnable are defined, to specify the types expected in the collections. 
https://github.com/jitsi/jitsi-stats/blob/master/src/main/java/org/jitsi/stats/media/AbstractStatsPeriodicRunnable.java#L107

It seems more clear for the expected result for the abstract methods, WDYT?